### ISSUE TITLE
The publish ios command fails

### ISSUE THREAD

**Issue by vchimev** [NativeScript/nativescript-cli#2158]
Command:
```
node_modules/.bin/tns publish ios <user> <pass> --path TNSAppTest --log trace
```

It seems there is `plugins-debug.xcconfig`, but no `plugins-release.xcconfig`:
```
mcsofnsbuild02:ios nsbuilduser$ pwd
/Users/nsbuilduser/workspace/tns-publish-iOS-AppStore-master/TNSAppTest/platforms/ios
mcsofnsbuild02:ios nsbuilduser$ ls -al
total 16
drwxr-xr-x   7 nsbuilduser  wheel  238 Oct 26 14:24 .
drwxr-xr-x   3 nsbuilduser  wheel  102 Oct 26 14:08 ..
-rw-r--r--   1 nsbuilduser  wheel   92 Oct 26 14:24 .nsprepareinfo
drwxr-xr-x  10 nsbuilduser  wheel  340 Oct 26 14:24 TNSAppTest
drwxr-xr-x   4 nsbuilduser  wheel  136 Oct 26 14:09 TNSAppTest.xcodeproj
drwxr-xr-x  14 nsbuilduser  wheel  476 Oct 26 14:09 internal
-rw-r--r--   1 nsbuilduser  wheel  131 Oct 26 14:24 plugins-debug.xcconfig
```
If I add `plugins-release.xcconfig`, the command manages to publish successfully.

Xcode version:
```
$ xcodebuild -version
Xcode 8.0
Build version 8A218a
```

Output:
```
Project successfully prepared (iOS)
AfterHookName for command prepare is after-prepare
spawn: xcodebuild "archive" "-archivePath" "/Users/nsbuilduser/workspace/tns-publish-iOS-AppStore-master/TNSAppTest/platforms/ios/build/archive/TNSAppTest.xcarchive" "-project" "/Users/nsbuilduser/workspace/tns-publish-iOS-AppStore-master/TNSAppTest/platforms/ios/TNSAppTest.xcodeproj" "-scheme" "TNSAppTest"
User defaults from command line:
    IDEArchivePathOverride = /Users/nsbuilduser/workspace/tns-publish-iOS-AppStore-master/TNSAppTest/platforms/ios/build/archive/TNSAppTest.xcarchive

=== BUILD TARGET TNSAppTest OF PROJECT TNSAppTest WITH CONFIGURATION Release ===

Check dependencies
build-release.xcconfig line 2: Unable to find included file "../plugins-release.xcconfig"
Signing for "TNSAppTest" requires a development team. Select a development team in the project editor.
Code signing is required for product type 'Application' in SDK 'iOS 10.0'

** ARCHIVE FAILED **


The following build commands failed:
	Check dependencies
(1 failure)
Error: Command xcodebuild failed with exit code 65
    at FiberFuture.wait (/Users/nsbuilduser/workspace/tns-publish-iOS-AppStore-master/node_modules/fibers/future.js:566:15)
    at /Users/nsbuilduser/workspace/tns-publish-iOS-AppStore-master/node_modules/nativescript/lib/common/errors.js:128:33
    at Function.settle (/Users/nsbuilduser/workspace/tns-publish-iOS-AppStore-master/node_modules/fibers/future.js:249:26)
    at FiberFuture.wait (/Users/nsbuilduser/workspace/tns-publish-iOS-AppStore-master/node_modules/fibers/future.js:565:10)
    at /Users/nsbuilduser/workspace/tns-publish-iOS-AppStore-master/node_modules/nativescript/lib/common/services/commands-service.js:84:106
    at Function.settle (/Users/nsbuilduser/workspace/tns-publish-iOS-AppStore-master/node_modules/fibers/future.js:249:26)
    at FiberFuture.wait (/Users/nsbuilduser/workspace/tns-publish-iOS-AppStore-master/node_modules/fibers/future.js:565:10)
    at /Users/nsbuilduser/workspace/tns-publish-iOS-AppStore-master/node_modules/nativescript/lib/common/yok.js:208:124
    at Function.settle (/Users/nsbuilduser/workspace/tns-publish-iOS-AppStore-master/node_modules/fibers/future.js:249:26)
    at FiberFuture.wait (/Users/nsbuilduser/workspace/tns-publish-iOS-AppStore-master/node_modules/fibers/future.js:565:10)
    at /Users/nsbuilduser/workspace/tns-publish-iOS-AppStore-master/node_modules/nativescript/lib/common/services/commands-service.js:51:55
    at Function.settle (/Users/nsbuilduser/workspace/tns-publish-iOS-AppStore-master/node_modules/fibers/future.js:249:26)
    at FiberFuture.wait (/Users/nsbuilduser/workspace/tns-publish-iOS-AppStore-master/node_modules/fibers/future.js:565:10)
    at /Users/nsbuilduser/workspace/tns-publish-iOS-AppStore-master/node_modules/nativescript/lib/common/errors.js:128:33
    at Function.settle (/Users/nsbuilduser/workspace/tns-publish-iOS-AppStore-master/node_modules/fibers/future.js:249:26)
    at FiberFuture.wait (/Users/nsbuilduser/workspace/tns-publish-iOS-AppStore-master/node_modules/fibers/future.js:565:10)
    - - - - -
    at ChildProcess.<anonymous> (/Users/nsbuilduser/workspace/tns-publish-iOS-AppStore-master/node_modules/nativescript/lib/common/child-process.js:91:38)
    at emitTwo (events.js:106:13)
    at ChildProcess.emit (events.js:191:7)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:215:12)
```

CC: @PanayotCankov, @tzraikov

**Comment by zakdances** [comment]
I'm using `Nativescript-CLI 2.4.0` and I'm still getting this error.

**Comment by bradmartin** [comment]
I think I ran into this last week. I switched from the CLI publish to an xcode workflow and I have to manually set the `PODS_ROOT` for release in the build settings for the project to `${SRCROOT}/Pods` to get a successful build.

**Comment by NathanWalker** [comment]
Confirmed, this is still an issue with 2.4.0 here.
Any estimate on when this fix will go out and would it be in 2.4.1? Thank you!

**Comment by tzraikov** [comment]
Hey @NathanWalker 2.4.1 is already public. Please, could you give it a try and confirm whether the issue still appears.

**Comment by bradmartin** [comment]
I can confirm it's still an issue. I just downgraded to 2.3.0 for another issue to test and I have the plugins-release.xcconfig in the platforms/ios folder after running `tns emulate ios` and then when opening in xcode I don't have to set the `PODS_ROOT` value anymore for release.

**Comment by NathanWalker** [comment]
@bradmartin Are you confirming it's still not working in `2.4.1`?

**Comment by bradmartin** [comment]
Fairly certain I was running latest. I'll try again later unless you want
to :)

**Comment by bradmartin** [comment]
Confirmed it's not there for me with latest release (not @next versions)

**Comment by bradmartin** [comment]
IGNORE ME*** Complete idiot about iOS 👋 
Everything works fine if you just run `tns build ios --release` stupid me. I only tried `tns emulate ios` which makes sense it only produces the debug files 😄

**Comment by vchimev** [comment]
@zakdances, @NathanWalker, @bradmartin,

the original issue here was that the `tns publish ios` command did not produce the *plugins-release.xcconfig* file, as it did not use `--release` option internally.

The issue was introduced and fixed prior to 2.4.0, so the `tns publish ios` command works as expected in 2.4.*.

For any other commands as `build`, `run`, etc.:
- in version 2.3 they used to prepare the project for both debug and release, no matter which flag has been specified,
- in version 2.4 this has been fixed as part of a major optimization of the project preparing (`tns prepare` command), so the respective flag `--debug` or `--release` should be passed to the commands.
NOTE: `--debug` is the default option when no flag specified.
